### PR TITLE
Check the domain before persisting cache artifact

### PR DESCRIPTION
**Pull request by tempoz**
Fixes buildbuddy-io/buildbuddy-internal#2463

### CODE REVIEW

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4939** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/4939" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/4939?utm_source=stack-comment).

File: server/build_event_protocol/build_event_handler/build_event_handler.go [hunk 1]
```diff
@@ -455,8 +457,10 @@ func (r *statsRecorder) handleTask(ctx context.Context, task *recordStatsTask) {
 			ctx := usageutil.WithLocalServerLabels(ctx)
 
 			fullPath := path.Join(task.invocationJWT.id, cacheArtifactsBlobstorePath, uri.Path)
-			if err := persistArtifact(ctx, r.env, uri, fullPath); err != nil {
-				log.CtxError(ctx, err.Error())
+			if urlutil.GetDomain(uri.Hostname()) == build_buddy_url.Domain() {
```
Comment (siggisim) [inline server/build_event_protocol/build_event_handler/build_event_handler.go:460]: Should we be comparing this to cache_api_url instead?

Comment (tempoz) [reply server/build_event_protocol/build_event_handler/build_event_handler.go:460]: mmm, probably, good catch.

File: server/build_event_protocol/build_event_handler/build_event_handler.go [hunk 2]
```diff
@@ -455,8 +457,10 @@ func (r *statsRecorder) handleTask(ctx context.Context, task *recordStatsTask) {
 			ctx := usageutil.WithLocalServerLabels(ctx)
 
 			fullPath := path.Join(task.invocationJWT.id, cacheArtifactsBlobstorePath, uri.Path)
-			if err := persistArtifact(ctx, r.env, uri, fullPath); err != nil {
-				log.CtxError(ctx, err.Error())
+			if urlutil.GetDomain(uri.Hostname()) == urlutil.GetDomain(cache_api_url.WithPath("").Hostname()) {
```
Comment (tylerwilliams) [inline server/build_event_protocol/build_event_handler/build_event_handler.go:460]: nit: add a comment?

// Only persist artifacts from caches that are hosted on the BuildBuddy domain.

**Review by tylerwilliams — APPROVED** [review]

File: server/build_event_protocol/build_event_handler/build_event_handler.go (hunk 2 above)
Comment (siggisim) [inline server/build_event_protocol/build_event_handler/build_event_handler.go:460]: How will this behave for on-prem folks that don't have `cache_api_url` set? Should we skip this check in that case?

Comment (tempoz) [reply server/build_event_protocol/build_event_handler/build_event_handler.go]: Done!

Comment (tempoz) [reply server/build_event_protocol/build_event_handler/build_event_handler.go]: Done!

**Pull request merged**